compiler/rustc_span/src/symbol.rs

```diff
@@ -1791,6 +1793,12 @@ symbols! {
         tool_lints,
         trace_macros,
         track_caller,
+        // reserved lang items for `default_auto_traits` feature
+        trait1,
+        trait2,
+        trait3,
+        trait4,
```

Comment:
I think we should just add the lang items under their matching name instead of reserving indexed items.

Reply:
In https://github.com/rust-lang/rust/pull/120706#discussion_r1482901925 above I suggested to use more generic naming to avoid bikeshedding specific traits.

I previously asked @Bryanskiy to also remove all libcore additions from this PR before merging (only keep the new traits in tests), but it looks like they are still in libcore.

Reply:
I must have misunderstood you. I'll fix it now.